### PR TITLE
Add support ReactiveDialog

### PR DESCRIPTION
**Pull request by prt2121**
Fix Missing support ReactiveDialog (https://github.com/ReactiveX/RxAndroid/issues/141).

### CODE REVIEW

File: rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java [hunk 1]
```diff
@@ -18,13 +18,13 @@
 
 import rx.Subscriber;
 
-final class SubscriberVault {
+public final class SubscriberVault {
```
Comment (hamidp) [inline rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java]: I'm not sure this should be public.

File: rxandroid-framework/src/main/java/rx/android/app/support/ReactiveDialog.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import rx.android.app.SubscriberVault;
+import rx.functions.Func1;
+
+/**
+ * Wrapper for DialogFragment that allows to observe on the result of the user interaction.
+ *
+ * @param <T> The type of data expected as return value from the fragment, can be boolean for confirmation dialogs,
+ *            or more complex for data input dialogs.
+ */
+public class ReactiveDialog<T> extends RxDialogFragment {
```
Comment (hamidp) [inline rxandroid-framework/src/main/java/rx/android/app/support/ReactiveDialog.java]: This is almost straight-up copy/pasted from ReactiveDialog, and it shares the same name.

The code can definitely be consolidated.

Comment (prt2121) [inline rxandroid-framework/src/main/java/rx/android/app/support/ReactiveDialog.java]: @hamidp sorry about that. I'll refactor it.

File: rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package rx.android.app;
+
+/**
+ * A wrapper for returning value from ReactiveDialog dialog.
+ *
+ * @param <V> The type of data expected as return value from the dialog.
+ */
+public final class ReactiveDialogResult<V> {
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java]: This should have `equals()` and `hashCode()` implementations. A `toString()` would also be nice.

File: rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    private final V value;
+    private final boolean canceled;
+
+    static <V> ReactiveDialogResult<V> asSuccess(V value) {
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java]: I think it's safe making both of these public.

File: rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java (hunk 1 above)
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java]: Indeed. Move it to `internal` if you need it in multiple packages.

File: rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogListener.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package rx.android.app;
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogListener.java]: Copyright

File: rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogObserver.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+package rx.android.app;
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogObserver.java]: Copyright

File: rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package rx.android.app;
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/ReactiveDialogResult.java]: Copyright

File: rxandroid-framework/src/main/java/rx/android/app/support/ReactiveDialog.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
+package rx.android.app.support;
```
Comment (JakeWharton) [inline rxandroid-framework/src/main/java/rx/android/app/support/ReactiveDialog.java]: License header

File: rxandroid-framework/src/test/java/rx/android/app/support/ReactiveDialogTest.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+import org.mockito.Mock;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowDialog;
+import rx.Observable;
+import rx.Observer;
+import rx.android.app.ReactiveDialogResult;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertNotNull;
```
Comment (JakeWharton) [inline rxandroid-framework/src/test/java/rx/android/app/support/ReactiveDialogTest.java]: You have an explicit import and a star import right above it. Please avoid star imports.

File: rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java (hunk 1 above)
Comment (prt2121) [inline rxandroid-framework/src/main/java/rx/android/app/SubscriberVault.java]: @hamidp @JakeWharton 
Thanks for reviewing. I moved the class to internal package.

**Comment by prt2121** [comment]
Anything I should fix? will this be merged?

**Comment by JakeWharton** [comment]
As part of #172 ReactiveDialog has been removed. There's far too many uses cases and feature requests for a component like this to make it viable in a core library. A reactive dialog would make a great standalone library that was very focused.

**Pull request closed without merging**